custom_components/opensprinkler/__init__.py

```diff
@@ -47,12 +51,25 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     url = entry.data.get(CONF_URL)
     password = entry.data.get(CONF_PASSWORD)
+    mqtt_option = entry.data.get(CONF_MQTT, DEFAULT_MQTT)
+
     try:
         controller = OpenSprinkler(url, password)
         controller.refresh_on_update = False
 
+        if mqtt_option:
+
+            async def on_message(message):
+                await hass.async_add_executor_job(controller.refresh)
+
+            await mqtt.async_subscribe(hass, MQTT_TOPIC, on_message)
+
         async def async_update_data():
             """Fetch data from OpenSprinkler."""
+
+            if mqtt_option:  # skip refresh
```

Comment:
Since mqtt attributes are a subset of the data we can’t shit this down completely can we? Maybe we can up the interval quite high but there are still things that aren’t sending mqtt updates that could hypothetically get out of sync I think.

Reply:
I think the main thing that would go out of sync is the enable/disable for controller/programs/stations. What should the interval be though? 1 min? 5 min? longer?

Reply:
set to 5 mins for now